x-pack/plugins/observability_solution/uptime/public/legacy_uptime/lib/helper/observability_integration/get_infra_href.ts

```diff
@@ -5,44 +5,47 @@
  * 2.0.
  */
 
+import { AssetDetailsLocator } from '@kbn/observability-shared-plugin/common';
 import { MonitorSummary, Ping } from '../../../../../common/runtime_types';
 import { addBasePath } from './add_base_path';
 import { buildHref } from './build_href';
 
 export const getInfraContainerHref = (
-  summary: MonitorSummary,
-  basePath: string
+  { state }: MonitorSummary,
+  locator?: AssetDetailsLocator
 ): string | undefined => {
-  const getHref = (value: string | string[] | undefined) => {
-    if (!value) {
-      return undefined;
-    }
-    const ret = !Array.isArray(value) ? value : value[0];
-    return addBasePath(
-      basePath,
-      `/app/metrics/link-to/container-detail/${encodeURIComponent(ret)}`
-    );
-  };
-  return buildHref(summary.state.summaryPings || [], (ping: Ping) => ping?.container?.id, getHref);
+  if (!locator) {
+    return undefined;
+  }
+
+  // Pick the first container id if one is available
+  const containerId = state.summaryPings?.[0]?.container?.id;
```

Comment:
is there any chance that `summaryPings` is not an array? If so, we might want to play on the safe side and make sure we're dealing with arrays, as the previous version was doing

```suggestion
  const containerId = castArray(state.summaryPings)[0]?.container?.id;
```